### DISCUSSION TITLE
TwigCS task does not check the diff, preventing commit on unchanged files

### DISCUSSION THREAD

**Discussion by jklmnop** [phpro/grumphp#931] [Ideas]
| Q               | A
| --------------- | ---
| Version         | 1.3.2
| Bug?            | no?
| New feature?    | yes
| Question?       | yes
| Documentation?  | yes/no
| Related tickets | n/a

We have a GrumPHP setup that utilizes the TwigCS task, but the task runs on the entire project instead of just the diff on pre-commit. This means that no commit will be passing until ALL twig issues are fixed.

It works as intended when run manually in a specific folder, though.

Should be noted that we're using lando for this project, but I do not suspect that is part of the issue. All other tasks work fine.

I've created a simple test project to illustrate this issue.

**Steps to reproduce:**

```sh
# Get the test project
$ git clone git@github.com:jklmnop/twigcstest.git

# Install dev dependencies
$ composer install

# Check what's broken on root
$ ./vendor/bin/grumphp run
```

Fix either twig file in `src/templates`, but not both. I fixed `two.twig` by adding `{{ something }}` to the file.

```sh
# Run GrumPHP:
git add -A && git commit -m"Test"
```

**Result:**

You should see TwigCS is unnecessarily checking `one.twig` which is unchanged and prevents the commit.

**Comment by cameronprince** [comment]
I can confirm this is happening and makes the plugin unusable...

**Comment by jklmnop** [comment]
> I can confirm this is happening and makes the plugin unusable...

@cameronprince If you need this functionality, you can update TwigCS to the latest version and use the patch in this comment to augment the GrumPHP TwigCS task.

https://github.com/phpro/grumphp/pull/920#issuecomment-926553588

Ultimately, though, this is going to require TwigCS to address the issue, as expressed in the other comments of that pull request.

Hope it helps. Note that it might be a little slow on the first run, but it should speed up after that.
